src/serverless/structure/serverless_parser.go

```diff
@@ -0,0 +1,317 @@
+}
+
+func (p *ServerlessParser) ParseFile(filePath string) ([]structure.IBlock, error) {
+	parsedBlocks := make([]structure.IBlock, 0)
+	template, err := ioutil.ReadFile(filePath)
+	if err != nil {
+		logger.Warning(fmt.Sprintf("There was an error processing the serverless template: %s", err))
+	}
+	err = yaml.Unmarshal([]byte(template), p.template)
+	if err != nil {
+		logger.Error(fmt.Sprintf("Unmarshal: %s", err), "SILENT")
+	}
+
+	if err != nil || template == nil {
+		logger.Error(fmt.Sprintf("There was an error processing the serverless template: %s", err), "SILENT")
+
+	}
+	//cfnStackTagsResource := p.template.Provider.CFNTags
+	functions := p.template.Provider.Functions
+	functionsMap := functions.(map[interface{}]interface{})
+	resourceNames := make([]string, 0)
+	var resourceNamesToLines map[string]*common.Lines
+	for funcName, _ := range functionsMap {
+		resourceNames = append(resourceNames, funcName.(string))
+	}
+	switch common.GetFileFormat(filePath) {
+	case common.YmlFileType.FileFormat, common.YamlFileType.FileFormat:
+		resourceNamesToLines = MapResourcesLineYAML(filePath, resourceNames)
+	default:
+		return nil, fmt.Errorf("unsupported file type %s", common.GetFileFormat(filePath))
+	}
+	minResourceLine := math.MaxInt8
+	maxResourceLine := 0
+	for _, funcName := range resourceNames {
+		var existingTags []tags.ITag
+		funcRange := functionsMap[funcName].(map[interface{}]interface{})
+		for key, val := range funcRange {
+			lines := resourceNamesToLines[funcName]
+			switch key {
+			case FunctionTagsAttributeName:
+				funcTags := val.(map[interface{}]interface{})
+				for tagKey, tagValue := range funcTags {
+					existingTags = append(existingTags, &tags.Tag{
+						Key:   tagKey.(string),
+						Value: tagValue.(string),
+					})
+				}
+				tagsLines := p.extractLines(filePath, lines, resourceNames)
+				rawBlock := funcRange
+				minResourceLine = int(math.Min(float64(minResourceLine), float64(lines.Start)))
+				maxResourceLine = int(math.Max(float64(maxResourceLine), float64(lines.End)))
+				slsBlock := &ServerlessBlock{
+					Block: structure.Block{
+						FilePath:          filePath,
+						ExitingTags:       existingTags,
+						RawBlock:          rawBlock,
+						IsTaggable:        true,
+						TagsAttributeName: FunctionTagsAttributeName,
+						Lines:             *lines,
+						TagLines:          tagsLines,
+					},
+					name: funcName,
+				}
+				parsedBlocks = append(parsedBlocks, slsBlock)
+			}
+			p.fileToResourcesLines[filePath] = common.Lines{Start: minResourceLine, End: maxResourceLine}
+		}
+	}
+	return parsedBlocks, nil
+}
+
+func (p *ServerlessParser) extractLines(filePath string, lines *common.Lines, resourceNames []string) common.Lines {
+	tagsLines := p.getTagsLines(filePath, lines, resourceNames)
+	return tagsLines
+}
+
+func (p *ServerlessParser) WriteFile(readFilePath string, blocks []structure.IBlock, writeFilePath string) error {
```

Comment:
@tronxd This function is almost identical duplication of cloudformation's `WriteFile` method, this implementation is better because it uses the common extensions. Perhaps extracting the yaml part to a method in `yaml_writer` is the way to go?